cypress/e2e/session_timeout.cy.js

```diff
@@ -0,0 +1,74 @@
+import './base.cy'
+
+describe('session timeout tests', () => {
+
+  beforeEach(()=>{
+    cy.clock();
+    cy.start()
+  });
+
+  it('session_timeout_dialog_success', () => {
+
+    cy.get('.govuk-grid-column-two-thirds > .govuk-button').click();
+    cy.get('#id_registrar_organisation').clear().type('WeRegister')
+    cy.get('#id_registrar_name').clear('Test');
+    cy.get('#id_registrar_phone').clear('01225672345');
+    cy.get('#id_registrar_email').clear('test@example.gov.uk');
+    cy.get('#id_submit').click();
+
+    // check if model_dialog doesn't exist
+    cy.get('#modal_dialog').should('not.be.visible');
+    // clock tick
+    cy.tick(15 * 60 * 1000);
+    cy.get('#modal_dialog').should('exist');
+    cy.get('#modal_dialog').should('be.visible');
+
+  });
+
+  it('session_timeout_click_continue_session', () => {
+
+    cy.get('.govuk-grid-column-two-thirds > .govuk-button').click();
+    cy.get('#id_registrar_organisation').clear().type('WeRegister')
+    cy.get('#id_registrar_name').clear('Test');
+    cy.get('#id_registrar_phone').clear('01225672345');
+    cy.get('#id_registrar_email').clear('test@example.gov.uk');
+    cy.get('#id_submit').click();
+
+    // clock tick
+    cy.tick(15 * 60 * 1000);
+    cy.get('#modal_dialog').should('exist');
+    cy.get('#modal_dialog').should('be.visible');
+
+    // Click the "Continue" button
+    cy.get('#session-continue').click();
+
+    // Check if the warning modal is hidden
+    cy.get('#modal_dialog').should('not.be.visible');
+
+  });
+
+  it('session_timeout_session_ended', () => {
```

Comment:
Did you try a test with multiple tabs? Since this is what justified the server-side timeout, we should have an example where the multi-tab handling works.